model/jpa/src/main/java/org/keycloak/models/jpa/entities/UserAttributeEntity.java

```diff
@@ -77,11 +80,23 @@ public void setName(String name) {
     }
 
     public String getValue() {
-        return value;
+        if (value != null && long_value != null) {
+            throw new IllegalStateException(String.format("User with id %s should not have set both `value` and `long_value`.", user.getId()));
```

Comment:
May I just suggest to add the attribute name in the message? Something like "User with id %s should not have set both `value` and `long_value`  for attribute %s", user.getId(), name ?